catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -1179,14 +1179,19 @@ protected UpdateResponse patchMetacards(List<MetacardChanges> metacardChanges)
       for (AttributeChange attributeChange : changeset.getAttributes()) {
         for (String id : changeset.getIds()) {
           List<String> values = attributeChange.getValues();
-          Metacard result = results.get(id).getMetacard();
+          Result result = results.get(id);
+          if (result == null) {
+            throw new NotFoundException(
+                "Result was not found or user did not have permission to view result");
```

Comment:
❓ Should we tell the user about potential permissions errors? I think we should make it as generic as possible. 

`Result not found.`

Reply:
SGTM 👍

Reply:
Actually I just realized I'm throwing this message on an exception, not actually returning it to the user. I think this detail will help with troubleshooting. Thoughts?

Reply:
@bellcc is right. Spark exception handlers are forwarding the message over HTTP. I'll change this and add a logging statement with more info.